rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -411,20 +411,39 @@ impl Controller {
         Ok(())
     }
 
-    fn reload_resource_list(&mut self) -> Result<()> {
-        let Some(session) = &self.session else {
-            tracing::warn!("got notified to update resources but there is no session");
-            return Ok(());
-        };
-        let resources = session.callback_handler.resources.load();
-        let actor_name = &self
-            .auth
-            .session()
-            .ok_or_else(|| anyhow!("connlib is signed in but auth module isn't"))?
-            .actor_name;
-        self.app
-            .tray_handle()
-            .set_menu(system_tray_menu::signed_in(actor_name, &resources))?;
+    /// Returns a new system tray menu
+    fn build_system_tray_menu(&self) -> Result<tauri::SystemTrayMenu> {
+        Ok(if let Some(auth_session) = self.auth.session() {
+            if let Some(connlib_session) = &self.session {
+                // TODO: Modify CallbackHandler so it can tell us whether the tunnel is really ready.
+                let tunnel_ready = true;
+                if tunnel_ready {
+                    // Signed in, tunnel ready
+                    let resources = connlib_session.callback_handler.resources.load();
+                    system_tray_menu::signed_in(&auth_session.actor_name, &resources)
+                } else {
+                    // Signed in, raising tunnel
+                    system_tray_menu::signing_in()
+                }
+            } else {
+                bail!("We have an auth session but no connlib session");
+            }
+        } else if self.auth.ongoing_request().is_ok() {
+            // Signing in, waiting on deep link callback
+            system_tray_menu::signing_in()
+        } else {
+            system_tray_menu::signed_out()
+        })
+    }
+
+    /// Builds a new system tray menu and applies it to the app
+    fn refresh_system_tray_menu(&self) -> Result<()> {
+        let menu = self.build_system_tray_menu().unwrap_or_else(|error| {
+            tracing::error!(?error, "couldn't build system tray menu");
+            // If there's an error, show the signed-out menu as a default.
+            system_tray_menu::signed_out()
```

Comment:
if we return `signed_out` by default in `build_system_tray_menu` we don't need the additional flow control here.